File: fax/sip-send/example-3/example-3.7.x.java
// Install the Java helper library from twilio.com/docs/java/install
import java.net.URI;
import java.net.URISyntaxException;

import com.twilio.Twilio;
import com.twilio.rest.fax.v1.Fax;
import com.twilio.type.PhoneNumber;
import com.twilio.type.Sip;

public class Example {
  // Find your Account Sid and Token at twilio.com/user/account
  public static final String ACCOUNT_SID = "AC850685e1d9b8c09dae0b938923dc0d42";
  public static final String AUTH_TOKEN = "your_auth_token";

  public static void main(String[] args) throws URISyntaxException {
    Twilio.init(ACCOUNT_SID, AUTH_TOKEN);

    String from = "Jack";
    String to = "sip:kate@example.com?hatchkey=4815162342;transport=TCP";
    URI mediaUrl = new URI("https://www.twilio.com/docs/documents/25/justthefaxmaam.pdf");
    Fax fax = Fax.creator(to, mediaUrl).setFrom(from).create();

    System.out.println(fax.getSid());
  }
}

File: fax/sip-send/example-3/example-3.7.x.rb
require 'net/http'
require 'json'

# Get your Account SID and Auth Token from twilio.com/console
# To set up environmental variables, see http://twil.io/secure
account_sid = ENV['TWILIO_ACCOUNT_SID']
auth_token = ENV['TWILIO_AUTH_TOKEN']

uri = URI('https://fax.twilio.com/v1/Faxes')

Net::HTTP.start(uri.host, uri.port, use_ssl: uri.scheme == 'https') do |http|
  request = Net::HTTP::Post.new uri.request_uri
  request.basic_auth(account_sid, auth_token)
  to = 'sip:kate@example.com?hatchkey=4815162342;transport=TCP'
  media_url = 'https://www.twilio.com/docs/documents/25/justthefaxmaam.pdf'
  request.body = URI.encode_www_form(to: to, from: 'Jack', media_url: media_url)

  response = http.request request

  puts JSON.parse(response.body)['sid']
end
